app.rb

```diff
@@ -27,27 +52,33 @@ class OpenidConnectRelyingParty < Sinatra::Base
       nonce: SecureRandom.hex,
       prompt: 'select_account',
     }.to_query
-
-    erb :index, locals: { authorization_url: authorization_url }
   end
 
-  get '/auth/result' do
-    token_response = token(params[:code])
-    userinfo_response = userinfo(token_response[:id_token])
-
-    erb :success, locals: { userinfo: userinfo_response }
+  def openid_configuration
+    @openid_configuration ||= begin
+      response = openid_configuration_response
```

Comment:
thoughts on memoizing `openid_configuration_response` so we don't need to save it as a var here? 

I do think that referencing it as `openid_configuration_response` in this method (rather than `response`), while verbose, is clearer.